packages/core/src/components/select/select.tsx

```diff
@@ -350,6 +349,12 @@ export class Select {
       this.value = this.selectedIndices;
     }
 
+    if (typeof this.items[0]?.value !== 'string') {
+      console.warn(
+        'Non string data types not supported and and will be removed in 3.0'
+      );
+    }
+
```

Comment:
Could be a potential undefined access if no items are provided. Maybe a better place to show this warning will be inside the forEach above.